source/lac/trilinos_sparsity_pattern.cc

```diff
@@ -791,7 +791,8 @@ namespace TrilinosWrappers
     if (nonlocal_graph.get() != nullptr)
       {
         if (nonlocal_graph->IndicesAreGlobal() == false &&
-            nonlocal_graph->RowMap().NumMyElements() > 0)
+            nonlocal_graph->RowMap().NumMyElements() > 0 &&
+            column_space_map->NumMyElements() > 0)
```

Comment:
Don't you want to check for `column_space_map->NumGlobalElements() > 0`? We have a switch for an empty column space map on the local process in line 811 below. See also #7833.

Reply:
I changed that.